src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Options/Option2`1.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Immutable;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Options
+{
+    /// <summary>
+    /// Marker interface for language specific options.
+    /// </summary>
+    internal interface ILanguageSpecificOption : IOptionWithGroup
+    {
+    }
+
+    /// <summary>
+    /// Marker interface for language specific options.
+    /// </summary>
+    internal interface ILanguageSpecificOption<T> : ILanguageSpecificOption
```

Comment:
This is an new interface type added to allow being used as a common base interface for `Option<T> and `Option2<T>`, so certain APIs can directly just use `ILanguageSpecificOption<T>` as a parameter rather then have two similar overloads. This is not always feasible though so is not used everywhere.